### DISCUSSION TITLE
How do I set a timeout limit for bitwuzla？

### DISCUSSION THREAD

**Discussion by xuyang-nudt** [bitwuzla/bitwuzla#58] [General] [answered]

**Comment by mpreiner** [comment]
Timeouts are currently not yet configurable via options or command line, but will be in the future. In the meantime you can use the `timeout` command in Linux, or resource limiting tools like [runlim](https://github.com/arminbiere/runlim) and [runexec](https://github.com/sosy-lab/benchexec/blob/main/doc/runexec.md).

**Comment by mpreiner** [comment, marked as answer]
@busyxu We now implemented options for timeouts: `-t/--time-limit` <ms> and `-T/--time-limit-per <ms>`. The former specifies an overall runtime limit of the solver (when used via the CLI), the latter specifies a time limit for check_sat() queries.
